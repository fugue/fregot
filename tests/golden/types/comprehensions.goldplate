{
    "command": "fregot",
    "arguments": ["repl", "--no-history-file"],
    "stdin": [
        ":type [x | nums = [1, 2, 3]; x = nums[_]]",
        ":type {x | nums = [1, 2, 3]; x = nums[_]}",
        ":type {k: v | nums = [1, 2, 3]; strs = ['one', 'two', 'three']; v = nums[_]; k = strs[_]}"
    ],
    "asserts": [
        {"stdout": "${GOLDPLATE_NAME}.stdout"},
        {"stderr": "${GOLDPLATE_NAME}.stderr"},
        {"exit_code": 0}
    ]
}
