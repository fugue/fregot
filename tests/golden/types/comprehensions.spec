{
    "command": "fregot",
    "arguments": ["repl", "--no-history-file"],
    "stdin": [
        ":type [x | nums = [1, 2, 3]; x = nums[_]]",
        ":type {x | nums = [1, 2, 3]; x = nums[_]}",
        ":type {k: v | nums = [1, 2, 3]; strs = ['one', 'two', 'three']; v = nums[_]; k = strs[_]}"
    ],
    "asserts": [
        {"stdout": "${SPEC_NAME}.stdout"},
        {"stderr": "${SPEC_NAME}.stderr"},
        {"exit_code": 0}
    ]
}
