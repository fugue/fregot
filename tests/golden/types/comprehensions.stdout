repl% [33m[[0mx [33m|[0m nums [33m=[0m [33m[[0m
  [31m1[0m[33m,[0m
  [31m2[0m[33m,[0m
  [31m3[0m
[33m][0m[33m;[0m x [33m=[0m nums[33m[[0m_[33m][0m[33m][0m [33m:[0m [34marray[0m[33m{[0m[34mnumber[0m[33m}[0m
repl% [33m{[0mx [33m|[0m nums [33m=[0m [33m[[0m
  [31m1[0m[33m,[0m
  [31m2[0m[33m,[0m
  [31m3[0m
[33m][0m[33m;[0m x [33m=[0m nums[33m[[0m_[33m][0m[33m}[0m [33m:[0m [34mset[0m[33m{[0m[34mnumber[0m[33m}[0m
repl% [33m{[0mk[33m:[0m v [33m|[0m nums [33m=[0m [33m[[0m
  [31m1[0m[33m,[0m
  [31m2[0m[33m,[0m
  [31m3[0m
[33m][0m[33m;[0m strs [33m=[0m [33m[[0m
  [31m"one"[0m[33m,[0m
  [31m"two"[0m[33m,[0m
  [31m"three"[0m
[33m][0m[33m;[0m v [33m=[0m nums[33m[[0m_[33m][0m[33m;[0m k [33m=[0m strs[33m[[0m_[33m][0m[33m}[0m [33m:[0m [34mobject[0m[33m{[0m[34mstring[0m[33m:[0m [34mnumber[0m[33m}[0m
repl% 